src/nvim/event/process.c

```diff
@@ -227,11 +228,10 @@ void process_stop(Process *proc) FUNC_ATTR_NONNULL_ALL
       abort();
   }
 
-  Loop *loop = proc->loop;
-  // Start a timer to periodically check if a signal should be sent to the job.
+  // Start a timer to verify that the job process terminated.
   ILOG("starting job kill timer");
-  uv_timer_start(&loop->children_kill_timer, children_kill_cb,
-                 KILL_TIMEOUT_MS, KILL_TIMEOUT_MS);
+  uv_timer_start(&proc->loop->children_kill_timer, children_kill_cb,
+                 KILL_TIMEOUT_MS, 0);
```

Comment:
Is the process loop here also the main loop? If so all processes use the same timer and a sequence of stopped processes could reset the timer without ever being killed.

Reply:
Based on libuv docs and my testing, each uv_timer_start call resets the timer. 

And the timer callback always iterates the processes. 

So I think this is fine?